### PR TITLE
Make show return "toList [1, 2, 3]" instead of "(Cons 1 (Cons 2 (Cons 3 Nil)))"

### PR DESCRIPTION
**Pull request by rightfold**
Fixes https://github.com/purescript/purescript-lists/issues/43

### CODE REVIEW

File: src/Data/List/Lazy.purs [hunk 1]
```diff
@@ -629,10 +629,7 @@ zip = zipWith Tuple
 --------------------------------------------------------------------------------
 
 instance showList :: (Show a) => Show (List a) where
-  show xs = "fromStrict (" ++ go (step xs) ++ ")"
-    where
-    go Nil = "Nil"
-    go (Cons x xs) = "Cons (" ++ show x ++ ") (" ++ go (step xs) ++ ")"
+  show xs = "toList " <> show (fromList xs :: Array a)
```
Comment (paf31) [inline src/Data/List/Lazy.purs]: Perhaps this is a good time to add a `take` with an ellipsis here to avoid trying to print infinite lazy lists?

**Comment by paf31** [comment]
:+1:

File: src/Data/List/Lazy.purs (hunk 1 above)
Comment (rightfold) [inline src/Data/List/Lazy.purs]: How many elements to show by default?

Comment (rightfold) [inline src/Data/List/Lazy.purs]: Let me add a unit test for that too before merging.

**Comment by garyb** [comment]
:+1:

File: src/Data/List/Lazy.purs [hunk 2]
```diff
@@ -629,10 +629,10 @@ zip = zipWith Tuple
 --------------------------------------------------------------------------------
 
 instance showList :: (Show a) => Show (List a) where
-  show xs = "fromStrict (" ++ go (step xs) ++ ")"
-    where
-    go Nil = "Nil"
-    go (Cons x xs) = "Cons (" ++ show x ++ ") (" ++ go (step xs) ++ ")"
+  show xs = "toList " <> show (fromList truncated :: Array a) <> ellipsis
+    where truncated = take 5 six
```
Comment (paf31) [inline src/Data/List/Lazy.purs]: Sorry, I'm going to be a pain here. I just thought about this a bit more and realized, this way, there's no way to `show` more than 5 elements. Maybe we should create another function `showPrefix :: forall a. (Show a) => Int -> List a -> String`? That way, `show` is unsafe on infinite lists, but things stay flexible.

**Comment by rightfold** [comment]
I'm redoing this since the array-specific functions are gone and there's conflicts.

**Pull request closed without merging**